### ISSUE TITLE
Add support for type application in the type parser

### ISSUE THREAD

**Issue by comonoidial** [wandernauta/viskell#177]
After PR #176 it looks like the parser is the only thing blocking the addition of functors and alike to the catalog.
The parser/type builder should be extended to support types like:
fmap :: Functor f => (a -> b) -> f a -> f b
sequence_ :: Monad m => [m a] -> m ()
traverse :: (Traversable t, Applicative f) => (a -> f b) -> t a -> f (t b)
first :: Arrow a => a b c -> a (b, d) (c, d)

**Comment by wandernauta** [comment]
I've linked a branch that adds rudimentary type application parsing to the type parser.

I think we also need something else, though - Viskell must be able to deduce that a list (any list) is a Functor, for example. I'm not yet seeing how that would fit in the catalog.

**Comment by comonoidial** [comment]
I think something along the following lines will do just fine:
```
       <class name="Functor">
        	<instance name="Maybe"/>
        	<instance name="[]"/>
        	<instance name="IO"/>
        </class>
```

**Comment by wandernauta** [comment]
Right, that looks reasonable. I'll give it a try.

**Comment by wandernauta** [comment]
Sweet, seems to work for Maybe:

![2015-10-04-163558_653x616_scrot](https://cloud.githubusercontent.com/assets/126832/10268348/0707495a-6ab6-11e5-98b5-9208123ea1c7.png)

`[]` is not a type constructor according to the parser, because it doesn't start with a capital letter. I'll try to correct that.

**Comment by wandernauta** [comment]
See #178.

**Comment by comonoidial** [comment]
Fixed by #178
